application/src/Site/BlockLayout/Asset.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace Omeka\Site\BlockLayout;
+
+use Omeka\Api\Representation\SiteRepresentation;
+use Omeka\Api\Representation\SitePageRepresentation;
+use Omeka\Api\Representation\SitePageBlockRepresentation;
+use Omeka\Entity\SitePageBlock;
+use Omeka\Stdlib\ErrorStore;
+use Laminas\Form\Element\Select;
+use Laminas\Form\Form;
+use Laminas\View\Renderer\PhpRenderer;
+
+class Asset extends AbstractBlockLayout
+{
+    public function getLabel() {
+        return 'Asset'; // @translate
+    }
+
+    public function prepareForm(PhpRenderer $view) 
+    {
+      $view->headScript()->appendFile($view->assetUrl('js/asset-form.js', 'Omeka'));
+    }
+    
+    public function onHydrate(SitePageBlock $block, ErrorStore $errorStore)
+    {
+        $data = $block->getData();
+        $block->setData($data);
+      }
+      
+    public function alignmentClassSelect(PhpRenderer $view, SitePageBlockRepresentation $block = null
+    ) {
+        $alignmentLabels = ['default', 'float left', 'float right', 'center'];
+        $alignmentValues = ['default', 'left', 'right', 'center'];
+        $alignment = $block ? $block->dataValue('alignment', 'default') : 'default';
+        $select = new Select('o:block[__blockIndex__][o:data][alignment]');
+        $select->setValueOptions(array_combine($alignmentValues, $alignmentLabels))->setValue($alignment);
+        $selectLabel = 'Alignment'; // @translate
+        $select->setAttributes(['title' => $selectLabel, 'aria-label' => $selectLabel]);
+        $html = '<div class="field"><div class="field-meta">';
+        $html .= '<label class="thumbnail-option" for="o:block[__blockIndex__][o:data][alignment]">' . $selectLabel . '</label></div>';
```

Comment:
`for` connects to an input using an `id`, not a name, so this doesn't properly associate with the input.

You'll need to set an id for the select (if you do it through the options/attributes system for the form element object, you can just print the label with the form label helper too and it will do the proper `for` for you).

Reply:
`thumbnail-option` class?

Reply:
> `for` connects to an input using an `id`, not a name, so this doesn't properly associate with the input.

I modeled this off the [alignment option used in Media Embed](https://github.com/omeka/omeka-s/blob/develop/application/src/Site/BlockLayout/Media.php#L53). They both seem to be generating an `id`, and associating properly. Am I doing something differently here?

Reply:
I tried one of these and it wasn't activating the input when you clicked the label... I'll have to double check.

On the general point of the ID being generated... I'll have to check that too... the only one I saw an ID on in the inspector I believe was actually the other input, the one you're hand-writing the markup for. Which is odd.